### ISSUE TITLE
node bootstrap needs improvements

### ISSUE THREAD

**Issue by errordeveloper** [weaveworks/eksctl#134]
Right now we have this:

https://github.com/weaveworks/eksctl/blob/c332060348833792c9a69f1ef71d1c8d76c3ffd7/pkg/eks/assets/1.10.0/2018-05-09/amazon-eks-nodegroup.yaml#L253-L282

There many ways how it can be improved, AWS CLI is probably not need and we could just writ out override units instead of using sed.

We would need to fix this in order to fix #104.

**Comment by nckturner** [comment]
For the changes that require coordination with the AMI, we will need to make sure they still work in a cloudformation only environment, i.e. the AMI can't pick up a dependency on eksctl.

**Comment by errordeveloper** [comment]
I think first thing would be to try refactoring this to leverage cloudinit's ability to write files instead of using sed on pre-existing files...

**Comment by errordeveloper** [comment]
I think we have a fix coming along in #132.

We have [cloudconfig package](https://github.com/weaveworks/eksctl/tree/goformation/pkg/cloudconfig), and this is [how it is used](https://github.com/weaveworks/eksctl/blob/goformation/pkg/cfn/builder/nodegroup.go#L228):

```
func (n *nodeGroupResourceSet) makeUserData() (*gfn.StringIntrinsic, error) {
	const (
		configDir            = "/etc/eksctl/"
		kubeletDropInUnitDir = "/etc/systemd/system/kubelet.service.d/"
	)

	config := cloudconfig.New()

	scripts := []string{
		"get_metadata.sh",
		"get_credentials.sh",
	}
	files := map[string][]string{
		configDir: {
			"authenticator.sh",
			"kubeconfig.yaml",
		},
	}

	files[kubeletDropInUnitDir] = []string{"10-eksclt.al2.conf"}

	config.AddPackages("jq")
	config.AddCommand("pip", "install", "--upgrade", "awscli")

	// 	if ubuntu {
	// 		scripts = append(scripts, "ubuntu_prepare.sh")
	// 		config.AddRunCmd("cloud-init-per", "once", "rm", "-f", "/var/lib/cloud/scripts/per-instance/00-EKS-config.sh")
	// 	}

	for dir, fileNames := range files {
		for _, fileName := range fileNames {
			data, info, err := getAsset(fileName)
			if err != nil {
				return nil, err
			}
			config.AddFile(cloudconfig.File{
				Path:        dir + fileName,
				Content:     data,
				Permissions: fmt.Sprintf("o", uint(info.Mode())),
			})
		}
	}

	for _, scriptName := range scripts {
		data, _, err := getAsset(scriptName)
		if err != nil {
			return nil, err
		}
		config.RunScript(scriptName, data)
	}

	config.AddCommand("systemctl", "daemon-reload")
	config.AddCommand("systemctl", "restart", "kubelet")

	body, err := config.Encode()
	if err != nil {
		return nil, errors.Wrap(err, "encoding user data")
	}

	logger.Debug("user-data = %s", string(body))
	return gfn.NewString(string(body)), nil
}
```

For the record, I've tried an existing (package in Juju)[https://godoc.org/github.com/juju/juju/cloudconfig/cloudinit], but it was too complicated and dependencies were just a total mess.

**Comment by errordeveloper** [comment]
Fixed via #132.
